internal/appconfig/serde.go

```diff
@@ -66,7 +86,7 @@ func (c *Config) WriteToFile(filename string) (err error) {
 		}
 	}()
 
-	_, err = c.WriteTo(file)
+	_, err = c.WriteTo(file, filepath.Ext(filename))
```

Comment:
honest question: how does it work on Windows these days? is a filename like MYDATA.JSON possible ? thinking on uppercase extension names